Fix mac warnings related to visibility settings

Issue overview
--------------

Build logs are polluted on mac with warnings like these two:

```
ld: warning: direct access in function 'boost::cpp_regex_traits<char>::get_catalog_name()' from file '/Users/dev/.conan/data/boost/1.73.0/_/_/package/076aff20c42e87a34bf7b270c852b7953bc51cd2/lib/libboost_
regex.a(cpp_regex_traits.o)' to global weak symbol 'guard variable for boost::cpp_regex_traits<char>::get_catalog_name_inst()::s_name' from file 'src/utilities/CMakeFiles/openstudio_utilities.dir/bcl/BCLM
easure.cpp.o' means the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.                 
              
ld: warning: direct access in function 'boost::cpp_regex_traits<char>::get_catalog_name()' from file '/Users/dev/.conan/data/boost/1.73.0/_/_/package/076aff20c42e87a34bf7b270c852b7953bc51cd2/lib/libboost_
regex.a(cpp_regex_traits.o)' to global weak symbol 'boost::cpp_regex_traits<char>::get_catalog_name_inst()::s_name' from file 'src/utilities/CMakeFiles/openstudio_utilities.dir/bcl/BCLMeasure.cpp.o' means
 the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.
```


## Possible Solution

Probably related to CXX_FLAGS:

```
-fvisibility=hidden
-fvisibility-inlines-hidden
```


### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): macOS
 - Version of OpenStudio (if using an intermediate build, include SHA): develop 3.1.0-rc4

Update.  Further along the build it throws a fatal error so adding `-fvisibility=hidden -fvisibility-inlines-hidden` for mac builds will not work. I think we'll need to investigate settings at the package level.   

```
gmake[2]: *** No rule to make target 'src/utilities/idd/SurfaceProperty_ExposedFoundationPerimeter_FieldEnums.hxx', needed by 'src/energyplus/CMakeFiles/openstudio_energyplus.dir/ForwardTranslator/ForwardTranslateSurfacePropertyExposedFoundationPerimeter.cpp.o'.  Stop.
gmake[2]: *** Waiting for unfinished jobs....
Building CXX object src/energyplus/CMakeFiles/openstudio_energyplus.dir/ForwardTranslator/ForwardTranslateSurfacePropertyConvectionCoefficientsMultipleSurface.cpp.o
/Users/jenkins/git/OpenStudioFull/develop/src/energyplus/ForwardTranslator/ForwardTranslateSurfaceControlMovableInsulation.cpp:39:10: fatal error: 'utilities/idd/SurfaceControl_MovableInsulation_FieldEnums.hxx' file not found
#include <utilities/idd/SurfaceControl_MovableInsulation_FieldEnums.hxx>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [src/energyplus/CMakeFiles/openstudio_energyplus.dir/build.make:4589: src/energyplus/CMakeFiles/openstudio_energyplus.dir/ForwardTranslator/ForwardTranslateSurfaceControlMovableInsulation.cpp.o] Error 1
gmake[2]: *** No rule to make target 'src/utilities/idd/OS_PlantEquipmentOperation_OutdoorWetBulbDifference_FieldEnums.hxx', needed by 'src/model/CMakeFiles/openstudio_model.dir/PlantEquipmentOperationOutdoorWetBulbDifference.cpp.o'.  Stop.
gmake[2]: *** Waiting for unfinished jobs....
Building CXX object src/model/CMakeFiles/openstudio_model.dir/PlantEquipmentOperationOutdoorDryBulbDifference.cpp.o
/Users/jenkins/git/OpenStudioFull/develop/src/energyplus/ForwardTranslator/ForwardTranslateSurface.cpp:56:10: fatal error: 'utilities/idd/BuildingSurface_Detailed_FieldEnums.hxx' file not found
#include <utilities/idd/BuildingSurface_Detailed_FieldEnums.hxx>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/ConanInstall.cmake
+++ b/ConanInstall.cmake
@@ -41,7 +41,7 @@ if(NOT CONAN_OPENSTUDIO_ALREADY_RUN)
     # The os is still Linux, the compiler is still GCC. But the GLIBC used is **way older**
     conan_add_remote(NAME openstudio-centos INDEX 0
       URL https://conan.openstudio.net/artifactory/api/conan/openstudio-centos)
- 
+
     # Pass `-D_GLIBCXX_USE_CXX11_ABI=0` to make sure it detects libstdc++ and not libstdc++1
     add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
     # Centos uses a different channel recipe for ruby
@@ -51,7 +51,7 @@ if(NOT CONAN_OPENSTUDIO_ALREADY_RUN)
     endif()
 
   else()
-    conan_add_remote(NAME nrel INDEX 0 
+    conan_add_remote(NAME nrel INDEX 0
       URL https://conan.openstudio.net/artifactory/api/conan/openstudio)
 
     if(BUILD_RUBY_BINDINGS OR BUILD_CLI)
@@ -83,6 +83,11 @@ if(NOT CONAN_OPENSTUDIO_ALREADY_RUN)
   endif()
 
   # TODO:  list(APPEND CONAN_OPTIONS "fmt:header_only=True")
+
+  if(APPLE)
+    # #4120 - global is the 'default' visibility in gcc/clang
+    list(APPEND CONAN_OPTIONS "boost:visibility=global")
+  endif()
 
   # You do want to rebuild packages if there's a newer recipe in the remote (which applies mostly to our own openstudio_ruby where we don't
   # bump the actual package version when we make changes) than the binaries were built with

--- a/src/utilities/core/PathHelpers.cpp
+++ b/src/utilities/core/PathHelpers.cpp
@@ -375,7 +375,7 @@ bool copyDirectory(const path& source, const path& destination) {
   for (const auto& file : openstudio::filesystem::recursive_directory_files(source)) {
     try {
       openstudio::filesystem::create_directories((destination / file).parent_path());
-      openstudio::filesystem::copy_file(source / file, destination / file);
+      openstudio::filesystem::copy_file(source / file, destination / file, openstudio::filesystem::copy_option::overwrite_if_exists);
     } catch (const std::exception&) {
       return false;
     }

